tests/cxx/quoted_includes_first.h

```diff
@@ -18,16 +18,14 @@ std::unique_ptr<IndirectSubDirClass> CreateIndirectSubDirClass();
 /**** IWYU_SUMMARY
 
 tests/cxx/quoted_includes_first.h should add these lines:
```

Comment:
It's kind of annoying that this testcase has so much structure seeing how its only purpose is to check that includes are emitted in the right order.

I'll give it a think and see if I can reduce it to something even simpler so it's not affected by this change (e.g. a `unique_ptr<int>` to force use of `<memory>`).

Also, I'm a little bit surprised that `std::unique_ptr<IndirectSubDirClass> CreateIndirectSubDirClass();` in a header now requires the full type of `IndirectSubDirClass`... That seems like something that might upset users who want to optimize by leaving the responsibility of the pointer type to callers. But they can just add an explicit `class IndirectSubDirClass;` in the header, and it should be respected?

Reply:
With [this PR](https://github.com/include-what-you-use/include-what-you-use/pull/1305) I'm hoping you wouldn't have to touch `quoted_includes_first.h` for this patch?

Reply:
Yes, it should. I just make the common policy for type components in all three cases of author-intent analysis (aliases, function return, "autocast"). Command line flags to suppress that policies can be later introduced.

Btw, in fact, returning `unique_ptr` requires pointed-to-type full info to be available at the call site, because it is needed for `~unique_ptr()`, in contrast to `shared_ptr`.

Reply:
> With https://github.com/include-what-you-use/include-what-you-use/pull/1305 I'm hoping you wouldn't have to touch quoted_includes_first.h for this patch?

I think, yes. Thank you!

Reply:
> Btw, in fact, returning unique_ptr requires pointed-to-type full info to be available at the call site, because it is needed for ~unique_ptr(), in contrast to shared_ptr.

Yeah, my instinct was also that it would be nice if the defining header required it. I think for real code it can get annoying if you have wide interfaces where one function happens to return a type that brings in a lot of headers, and that function is rarely used. See also issue 1297 for a similar thing (didn't want to link it, because it doesn't really have anything to do with this PR)

Reply:
There is currently a bug on the master branch. Consider:
```cpp
// --- GetClass.h

#include "Class.h"

#include <memory>

std::unique_ptr<Class> GetClass();


// --- User.cpp

#include "Class.h"
#include "GetClass.h"

void User() {
    GetClass();
}
```
IWYU suggests to remove `#include "Class.h"` from both `GetClass.h` and `User.cpp`. The code stops to compile after that, of course. This PR fixes the bug leaving `#include "Class.h"` in `GetClass.h`. Do you propose introducing of policy-suppression flags, which would push `#include` to the user sides from the decl sides, before merging this PR? Or, maybe, some different policy for return types?

Reply:
No, I don't think we need to worry about it now, I was just thinking out loud. https://github.com/include-what-you-use/include-what-you-use/blob/master/docs/WhatIsAUse.md#automatic-re-export-function-return-values clearly says headers declaring functions are responsible for their return type, unless they forward-declare to indicate user is responsible.